src/vaadin-radio-button-mixin.ts

```diff
@@ -0,0 +1,134 @@
+import { KeyboardClass } from '@vaadin/keyboard-mixin/keyboard-class.js';
+import { SlottedLabelMixin } from '@vaadin/slotted-label-mixin/slotted-label-mixin.js';
+import { RadioButtonMixinClass } from './vaadin-radio-button-mixin-class';
+
+export interface RadioButtonInterface {
+  value: string | null | undefined;
+}
+
+export type RadioButton = RadioButtonInterface &
+  DisabledStateInterface &
+  FocusVisibleInterface &
+  ControlStateInterface &
+  CheckedStateInterface &
+  ActiveStateClass &
+  KeyboardClass &
+  RadioButtonMixinClass;
+
+export const RadioButtonMixin = <T extends Constructor<LitElement>>(base: T): T & Constructor<RadioButton> => {
+  class RadioButton extends SlottedLabelMixin(
+    CheckedStateMixin(ControlStateMixin(FocusVisibleMixin(ActiveStateMixin(DisabledStateMixin(base)))))
+  ) {
+    /**
+     * Used for mixin detection because `instanceof` does not work with mixins.
+     */
+    static hasRadioButtonMixin = true;
+
+    /**
+     * Value of the radio button. Required when used in a radio group.
+     */
+    @property({ type: String }) value: string | null | undefined = 'on';
+
+    protected render() {
+      return html`
+        <label>
+          <span part="radio">
+            <input
+              type="radio"
+              .checked="${this.checked}"
+              ?disabled="${this.disabled}"
+              role="presentation"
+              @change="${this._onChange}"
+              tabindex="-1"
+            />
+          </span>
+
+          <span part="label">
+            <slot></slot>
+          </span>
+        </label>
+      `;
+    }
+
+    protected firstUpdated(props: PropertyValues) {
+      super.firstUpdated(props);
+
+      this.setAttribute('role', 'radio');
+
+      this.addEventListener('mouseup', event => this._onMouseUp(event));
+    }
+
+    /**
+     * Toggles the radio button, so that the native `change` event
+     * is dispatched. Overrides the `HTMLElement.prototype.click`.
+     * @protected
+     */
+    click() {
+      // If you change this block, please test manually that radio-button and
+      // radio-group still works ok on iOS 12/13 and up as it may cause
+      // an issue that is not possible to test programmatically.
+      // See: https://github.com/vaadin/vaadin-radio-button/issues/140
+      if (!this.disabled) {
+        this.focusElement.dispatchEvent(new MouseEvent('click'));
+      }
+    }
+
+    protected get focusElement() {
+      return this.renderRoot.querySelector('input') as HTMLInputElement;
+    }
+
+    protected _onChange(event: Event) {
+      this.checked = (event.target as HTMLInputElement).checked;
+
+      const changeEvent = new CustomEvent('change', {
+        detail: {
+          sourceEvent: event
+        },
+        bubbles: event.bubbles,
+        cancelable: event.cancelable
+      });
+      this.dispatchEvent(changeEvent);
+    }
+
+    protected _onKeyDown(event: KeyboardEvent) {
```

Comment:
Added JSDoc annotation `@see KeyboardMixin`. Unfortunately it does not seem to support "jump to definition" in VS Code at the moment, maybe it could be implemented in future.

Reply:
Weirdly my comment above seems to be in wrong order as I published my review including it after @web-padawan's comment above. Please check my comment above not to miss it.

Reply:
@Haprog the `KeyboardMixin` could be used explicitly. Will do.
It is a "deduping" mixin which is applied by `KeyboardDirectionMixin` internally.

Reply:
Ok, thanks. I didn't notice that.

Reply:
I'm fine with marking this as resolved. What about @yuriy-fix?

Reply:
Yeah, will resolve.